.github/workflows/rule-validate.yml

```diff
@@ -8,8 +8,8 @@ on:
   workflow_dispatch: {}
 
 concurrency:
-  group: ${{ github.ref }}
-  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true
```

Comment:
`github.ref` would be empty for on.push.branches events, so this would have already always been `true`. I'd also be fine having this always be `false`. I think it's not very relevant for rules where CI is fast and the rate of merges is low (and now we require CI to run on up to date branches anyways 🤷 )